CMakeLists.txt

```diff
@@ -1719,7 +1719,31 @@ endif()
 
 # Generate and use MCUboot related artifacts as needed.
 if(CONFIG_BOOTLOADER_MCUBOOT)
-  include(${CMAKE_CURRENT_LIST_DIR}/cmake/mcuboot.cmake)
+  get_property(signing_script_set
+    GLOBAL PROPERTY
+    signing_script SET
+    )
+
+  # If the signing_script property is not set, use the default MCUboot one included with Zephyr
+  if(${signing_script_set} EQUAL "0")
+    set_property(GLOBAL PROPERTY signing_script ${CMAKE_CURRENT_LIST_DIR}/cmake/mcuboot.cmake)
+  endif()
+endif()
+
+# Include signing script, if set
+get_property(signing_script_set
+  GLOBAL PROPERTY
+  signing_script SET
+  )
```

Comment:
I believe this `get_property(... SET)` is superfluous cause if it was unset before then you have just set the value above so no need to re-test whether it's set now.